src/routes/user.settings.route.js

```diff
@@ -0,0 +1,50 @@
+import express from 'express';
+import userController from '../controllers/user.controller';
+import verifyToken from '../middlewares/verify.token.middleware';
+import verifyAdmin from '../middlewares/admin.verify.middleware';
+import verifyUser from '../middlewares/verify.user.middleware';
+
+const router = express.Router();
+
+
+/**
+ * @swagger
+ *
+ * /user/role:
+ *  patch:
+ *    tags:
+ *      - Users
+ *    summary: Admin Can Update User Role And Permission
+ *    description: Admin Update User Role by his Email to Barefoot Nomad
+ *      and send data after updating
+ *    operationId: UpdateUserRole
+ *    requestBody:
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               role:
+ *                 type: string
+ *               email:
+ *                 type: string
+ *    produces:
+ *      - application/json
+ *    parameters:
+ *      - in: header
+ *        name: token
+ *        description: Returning User To Update users role
+ *        required: true
+ *        type: string
+ *        format: int64
+ *    responses:
+ *      '200':
+ *        description: successful operation
+ *      '404':
+ *        description: Unsuccessful User not found
+ *      '500':
+ *        description: Internal Server error
+ * */
+router.patch('/settings/updaterole', verifyToken.headerToken, verifyUser, verifyAdmin, userController.updateRole);
```

Comment:
@Jajabenit250 It is repetitive and unnecessary to use verbs in a route that are already supported by the HTTP method. For instance, we know that patch is used to partially update a given resource, so no need to use */updaterole* in the route. Additionally, I had suggested using this route */users/:userID/settings/*. Kindly update accordingly.

Reply:
@bdushimi  thanks for the feedback. i need clarity on what userId will stand for? while we are getting user email in body? in this route users/:userId/settings and again how will i know if am only changing role settings of the user ?